### PR TITLE
Add task to build mining pipeline fat JAR

### PR DESCRIPTION
**Pull request by FWDekker**
Adds a task to build the mining pipeline fat JAR.

### CODE REVIEW

File: modules/application/build.gradle [hunk 1]
```diff
@@ -22,3 +22,16 @@ dependencies {
 
     compile group: "commons-cli", name: "commons-cli", version: commonsCliVersion
 }
+
+// Distribution
+inspectClassesForKotlinIC.enabled = false
```
Comment (casperboone) [inline modules/application/build.gradle]: What does this line do?

Comment (gandreadis) [reply modules/application/build.gradle]: Necessary due to a bug in the newest Kotlin version, see https://youtrack.jetbrains.com/issue/KT-24956

Comment (casperboone) [reply modules/application/build.gradle]: In that case, can we put that in a comment?

Comment (gandreadis) [reply modules/application/build.gradle]: @FWDekker

Comment (FWDekker) [reply modules/application/build.gradle]: Yes

**Pull request merged**